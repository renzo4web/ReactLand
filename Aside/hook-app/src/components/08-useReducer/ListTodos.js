import TodoListItem from "./TodoListItem";

const ListTodos = ({
  todos = [],
  onRemove,
  onCompleted,
}) => {
  return (
    <ul className="list-group mt-3">
      {todos.map((todo) => (
        <TodoListItem
          key={`${todo}-${todo.id}`}
          k={`${todo}-${todo.id}`}
          item={todo}
          onRemove={onRemove}
          onCompleted={onCompleted}
        />
      ))}
    </ul>
  );
};

export default ListTodos;
