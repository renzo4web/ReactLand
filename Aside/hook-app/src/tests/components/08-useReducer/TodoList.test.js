import { shallow } from "enzyme";
import "jest";
import React from "react";
import ListTodos from "../../../components/08-useReducer/ListTodos";
import { demoTodos } from "../../fixtures/demoTodos";

describe("<TodoList /> Tests", () => {
  const onRemove = jest.fn();
  const onCompleted = jest.fn();

  const wrapper = shallow(
    <ListTodos
      todos={demoTodos}
      onRemove={onRemove}
      onCompleted={onCompleted}
    />
  );
  test("should render properly ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render the todos received", () => {
    expect(wrapper.find("TodoListItem").length).toBe(
      demoTodos.length
    );

    expect(
      wrapper.find("TodoListItem").at(0).prop("onRemove")
    ).toEqual(expect.any(Function));
  });
});
