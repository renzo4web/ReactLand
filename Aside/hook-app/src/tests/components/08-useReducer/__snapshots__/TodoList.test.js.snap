// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> Tests should render properly  1`] = `
<ul
  className="list-group mt-3"
>
  <TodoListItem
    item={
      Object {
        "completed": false,
        "id": 1,
        "todo": "Learn React",
      }
    }
    k="[object Object]-1"
    key="[object Object]-1"
    onCompleted={[MockFunction]}
    onRemove={[MockFunction]}
  />
  <TodoListItem
    item={
      Object {
        "completed": false,
        "id": 2,
        "todo": "Learn CSS",
      }
    }
    k="[object Object]-2"
    key="[object Object]-2"
    onCompleted={[MockFunction]}
    onRemove={[MockFunction]}
  />
</ul>
`;
